### PR TITLE
N3 parser: do not create formulas if the Turtle mode is activated

### LINKED ISSUE #1141: Turtle parser fails when a store without formula support is used
The Turtle parser fails with the following error when a `Store` without formula support is used (i.e. with `formula_aware = False`).

Errror:
```
    g.parse("test.ttl", format="ttl")
venv/lib/python3.8/site-packages/rdflib/graph.py:1549: in parse
    context.parse(source, publicID=publicID, format=format, **args)
venv/lib/python3.8/site-packages/rdflib/graph.py:1078: in parse
    parser.parse(source, self, **args)
venv/lib/python3.8/site-packages/rdflib/plugins/parsers/notation3.py:1884: in parse
    p = SinkParser(sink, baseURI=baseURI, turtle=turtle)
venv/lib/python3.8/site-packages/rdflib/plugins/parsers/notation3.py:417: in __init__
    self._formula = store.newFormula()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <rdflib.plugins.parsers.notation3.RDFSink object at 0x7fb6fe2f1a60>

    def newFormula(self):
>       assert self.graph.store.formula_aware
E       AssertionError

venv/lib/python3.8/site-packages/rdflib/plugins/parsers/notation3.py:1712: AssertionError
```

Code:
```python
g = Graph(store="OxMemory")
g.parse("test.ttl", format="ttl")
```

The store is provided by [oxrdflib](https://github.com/oxigraph/oxrdflib).

### PR DESCRIPTION
**Pull request by Tpt**
Fixes #1141

## Proposed Changes
  - Do not attempt to build a default formula if the Turtle mode is activated

### CODE REVIEW

**Comment by Tpt** [comment]
Hi @ashleysommer! Thank you for your feedback and sorry for the buggy change.

I have added a test to ensure there is no regression.
Indeed, having a null context was creating a blank node Id collision problem.
I have fixed it. Hopefully all tests should pass now.

File: rdflib/plugins/parsers/notation3.py [hunk 1]
```diff
@@ -1745,7 +1746,7 @@ def newBlankNode(self, arg=None, uri=None, why=None):
             return arg.newBlankNode(uri)
         elif isinstance(arg, Graph) or arg is None:
             self.counter += 1
-            bn = BNode("n" + str(self.counter))
+            bn = BNode("n%sb%s" % (self.uuid, self.counter))
```
Comment (ashleysommer) [inline rdflib/plugins/parsers/notation3.py:1749]: I'm going to have to look into this a little deeper. I don't know if adding a UUID to the RDFSink is the best way to approach this.

File: test/test_issue1141.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    Tests is Turtle and TriG parsing works with a store with or without formula support
+    """
+    def test_issue_1141(self):
+        file = b"@prefix : <http://example.com/> . :s :p :o ."
+
+        for format in ("turtle", "trig"):
+            # with formula
+            graph = Graph()
+            self.assertTrue(graph.store.formula_aware)
+            graph.load(BytesIO(file), format=format)
```
Comment (ashleysommer) [inline test/test_issue1141.py:20]: Just FYI, in RDFLib 5.1.0 the `graph.load()` method will be deprecated as part of the Graph API cleanup process, and in 6.0.0 `load()` will be removed.
You will need to use `graph.parse()` instead. In this case where `file` is a string with the RDF data, use `graph.parse(data=file)`.

**Comment by ashleysommer** [comment]
@Tpt this PR will require further and more thorough review by the maintainers team. The reason being this section of code you're changing is literally 15 to 20 years old, hasn't changed in a very long time, and used by hundreds of thousands of people every day for critical processes. Every tiny little change here could cause unforeseen problems in places we haven't thought of.

File: rdflib/plugins/parsers/notation3.py (hunk 1 above)
Comment (Tpt) [reply rdflib/plugins/parsers/notation3.py:1749]: UUID usage is what is currently done internally by the formula blank node generator

File: test/test_issue1141.py (hunk 2 above)
Comment (Tpt) [reply test/test_issue1141.py:20]: Ok ! Thank you! I'll fix it if the global PR design is accepted

**Comment by Tpt** [comment]
Thank you for the review! Indeed, there is a significant risk of unforeseen problem. I found the blank node ID generation one but there might be others.

**Comment by ashleysommer** [comment]
@Tpt
Sorry about the long delay in reviewing this.
I've done some extensive testing on this today, and I'm happy to say that its all good!
We can merge this, and this will open up the possibility of having more simplified store backends as Turtle Parser targets.

**Comment by Tpt** [comment]
@ashleysommer Thank you so much!

**Pull request merged**